backend/test/integration/healthz_api_test.go

```diff
@@ -43,10 +44,27 @@ func (s *HealthzApiTest) SetupTest() {
 			glog.Exitf("Failed to initialize test. Error: %v", err)
 		}
 	}
+
 	s.namespace = *namespace
-	clientConfig := test.GetClientConfig(*namespace)
+
+	var newHealthzClient func() (*api_server.HealthzClient, error)
+
+	if *isKubeflowMode {
+		s.resourceNamespace = *resourceNamespace
+
+		newHealthzClient = func() (*api_server.HealthzClient, error) {
+			return api_server.NewKubeflowInClusterHealthzClient(s.namespace, *isDebugMode)
+		}
+	} else {
+		clientConfig := test.GetClientConfig(*namespace)
+
+		newHealthzClient = func() (*api_server.HealthzClient, error) {
+			return api_server.NewHealthzClient(clientConfig, *isDebugMode)
+		}
+	}
+
 	var err error
-	s.healthzClient, err = api_server.NewHealthzClient(clientConfig, false)
+	s.healthzClient, err = newHealthzClient()
 	if err != nil {
 		glog.Exitf("Failed to get healthz client. Error: %v", err)
 	}
```

Comment:
NIT: as we don't use `newHealthzClient` function anywhere else, consider simplifying this to something more readable:
```golang
var err error
if *isKubeflowMode {
    s.resourceNamespace = *resourceNamespace
    s.healthzClient, err = api_server.NewKubeflowInClusterHealthzClient(s.namespace, *isDebugMode)
} else {
    clientConfig := test.GetClientConfig(*namespace)
    s.healthzClient, err = api_server.NewHealthzClient(clientConfig, *isDebugMode)
}
if err != nil {
    glog.Exitf("Failed to get healthz client. Error: %v", err)
}
```

Reply:
Sounds good. Can I fix it in another PR, because all other tests are using the same style which need to be updated as well.

Reply:
Offline discussion with @gkcalat: will not fix this in order to keep the style consistent with other test file.